internal/workload/manager.go

```diff
@@ -450,10 +502,36 @@ func (w *WorkloadManager) toPod(workload *models.Workload) (*v1.Pod, error) {
 	return &pod, nil
 }
 
+func (w *WorkloadManager) podConfigurationModified(manifestPath string, podYaml []byte, authPath string, auth string) bool {
+	return w.podModified(manifestPath, podYaml) || w.podAuthModified(authPath, auth)
+}
+
 func (w *WorkloadManager) podModified(manifestPath string, podYaml []byte) bool {
 	file, err := ioutil.ReadFile(manifestPath)
 	if err != nil {
 		return true
 	}
 	return bytes.Compare(file, podYaml) != 0
 }
+
+func (w *WorkloadManager) getAuthFilePathIfExists(workloadName string) string {
+	authFilePath := w.getAuthFilePath(workloadName)
+	if _, err := os.Stat(authFilePath); err != nil {
+		return ""
+	}
+	return authFilePath
+}
+
+func (w *WorkloadManager) podAuthModified(authPath string, auth string) bool {
+	if _, err := os.Stat(authPath); err != nil {
```

Comment:
Is there a need to check the error to read the file only for 'file not found' like ?

Reply:
I didn't think so - if there is some disk/fs problem, it will be handled here: https://github.com/jakub-dzon/k4e-device-worker/pull/53/files#diff-8be2736004b01f09b6ffef40a12d93f4f3f99cf3c84bc32e78100c086122cc44R162